lib/src/open_api/spec.dart

```diff
@@ -528,6 +528,8 @@ Map<String, dynamic> _formatSpecFromJson(
     if (m.containsKey('type')) {
       if (m['type'] == 'string' && m.containsKey('enum')) {
         m['type'] = 'enumeration';
+        final d = m['description'] ?? '';
+        m['description'] = '$d\nPossible values: ${m['enum']}';
```

Comment:
@davidmigloz yes, I would probably move this to the generator. Although, I do have a question: 

Why are we re-documenting `enum` values in the doc strings. This seems redundant. Simply clicking on the type should take you to the enum definition (lives in its own file) that you can quickly inspect to see the values. I think this adds a lot of extra documentation that may be unnecessary and distracting, no?

Reply:
Currently, if the enum is defined as an inner schema, it is generated with type `String`.

Eg. 
```
schemas:
  CreateCompletionRequest:
    type: object
    description: Request object for the Create completion endpoint.
    properties:
      model:
        description: &model_description |
          ID of the model to use. You can use the [List models](https://platform.openai.com/docs/api-...
        type: string
        enum:
          [
            "babbage-002",
            "davinci-002",
            "gpt-3.5-turbo-instruct",
            "text-davinci-003",
            "text-davinci-002",
            "text-davinci-001",
            "code-davinci-002",
            "text-curie-001",
            "text-babbage-001",
            "text-ada-001",
          ]
```

It will be converted into:
```dart
/// Factory constructor for CreateCompletionRequest
const factory CreateCompletionRequest({
  /// ID of the model to use. You can use the [List models](https://platform.openai.com/doc...
  ///
  /// Possible values: [babbage-002, davinci-002, gpt-3.5-turbo-instruct, text-davinci-003,...
  required String model,
})
```

So, in this case, documenting the possible values is nice. I agree, that when using an actual enum it is redundant.

I've moved that logic to the generator and only for this specific case. If you define the enum as a separate schema (which by the way is broken at the moment), it won't document the enum values.

Let me know if that works for you.

Reply:
> If you define the enum as a separate schema (which by the way is broken at the moment)

What do you mean by this? Maybe I am misunderstanding, but I generate enum schemas all the time from defined component schemas. Here is an example:

```json
"components": {
  "schemas": {
    "Task": {
      "title": "Task",
      "enum": [
        "pending",
        "completed",
        "failed"
      ],
      "type": "string",
      "description": "An enumeration."
    },
  }
}
```

```dart
enum Task {
  @JsonValue('pending')
  inProgress,
  @JsonValue('completed')
  canceled,
  @JsonValue('failed')
  failed,
}

```

Reply:
I haven't dug into it yet, but if instead of this:
```yaml
CreateCompletionRequest:
  type: object
  description: Request object for the Create completion endpoint.
  properties:
    model:
      description: ...
      type: string
      enum:
        [
          "babbage-002",
          "davinci-002",
          "gpt-3.5-turbo-instruct",
          "text-davinci-003",
          "text-davinci-002",
          "text-davinci-001",
          "code-davinci-002",
          "text-curie-001",
          "text-babbage-001",
          "text-ada-001",
        ]
```

I move it to its own schema:
```yaml
CompletionModel:
  type: string
  enum:
    [
      "babbage-002",
      "davinci-002",
      "gpt-3.5-turbo-instruct",
      "text-davinci-003",
      "text-davinci-002",
      "text-davinci-001",
      "code-davinci-002",
      "text-curie-001",
      "text-babbage-001",
      "text-ada-001",
    ]
CreateCompletionRequest:
  type: object
  description: Request object for the Create completion endpoint.
  properties:
    model:
      $ref: "#/components/schemas/CompletionModel"
```

I get the following exception:
```
Unhandled exception:
'package:openapi_spec/src/open_api/index.freezed.dart': Failed assertion: line 10478 pos 13: 'values == null || ref == null': Cannot define both values and ref
```

I'll look into it in a bit.

Reply:
I think I've fixed it in the latest commit. I don't think I've broken anything but take a look.

Reply:
The issue seems to be that when dereferencing an enum the new `_SchemaEnum` will have both `ref` and `values`, which is not allowed by the assert. 

I've removed the assert (which I guess it's not ideal), do you have a better suggestion?